helix-core/src/test/java/org/apache/helix/messaging/handling/TestResourceThreadpoolSize.java

```diff
@@ -113,6 +113,35 @@ public void TestThreadPoolSizeConfig() {
     }
   }
 
+  @Test
+  public void testBatchMessageThreadPoolSize() throws InterruptedException {
```

Comment:
Good suggestion! I tested it manually add the delay in batch message handle, it passed! Unfortunately, we does not have proper way to have a test BatchMessageHandler. So I just add a delay for state transitions. Hope this help.